source/extensions/access_loggers/grpc/http_grpc_access_log_impl.cc

```diff
@@ -158,7 +166,12 @@ void HttpGrpcAccessLog::emitLog(const Http::RequestHeaderMap& request_headers,
     }
   }
 
-  tls_slot_->getTyped<ThreadLocalLogger>().logger_->log(std::move(log_entry));
+  if (critical_log_filter_ && critical_log_filter_->evaluate(stream_info, request_headers,
+                                                             response_headers, response_trailers)) {
+    tls_slot_->getTyped<ThreadLocalLogger>().logger_->criticalLog(std::move(log_entry));
```

Comment:
Given that you already created a dedicated reliable logger class `GrpcCriticalAccessLogger`, I think adding also the `criticalLog` should not be necessary. I think it would be cleaner to keep logging API unchanged and just add `reliable_logger_` member to the `ThreadLocalLogger` class